exercises/react/src/example.c

```diff
@@ -0,0 +1,210 @@
+#include <stdlib.h>
+#include "react.h"
+
+enum cell_kind {
+   kind_input,
+   kind_compute1,
+   kind_compute2,
+};
+
+struct child {
+   struct cell *cell;
+   struct child *next;
+};
+
+struct cb {
+   callback_id id;
+   callback f;
+   void *obj;
+   struct cb *next;
+};
+
+struct reactor {
+   struct child *first;
+   struct child *last;
```

Comment:
It helps because if the list is populated, then we do `r->last->next = child` (since that `r->last` is a read it's not write-only)

It's true that it'd be possible to elide this field, and just find `last` by traversing the list - though saving this memory would mean more time (and also more code! since we have to write the code to traverse and find last).

An alternative question to ask is: "Why must newly-created cells be placed at the *tail* of the list? Could we delete this *last* field if we put newly-created cells at the *head* of the list?"

I think the answer to this question is related to `destroy_reactor` - I think I needed to destroy a cell before all its dependents, but now I think that might not be true, since I can just destroy the `struct child` object, only destroy the `cell` when I come to it on the list.

so, I'll see if I can keep only one end of this list, or report back with a reason why I must keep both ends.